File: Arrays/Easy/rotate_by_k_places.py
def rotate_right(nums, k):
    n = len(nums)
    ans = [0] * n
    for i in range(n):
        ans[(i + k) % n] = nums[i]
        
    for i in range(n):
        nums[i] = ans[i]  # update in-place


def rotate_left(nums, k):
    n = len(nums)
    ans = [0] * n
    for i in range(n):
        ans[(i + n -k ) % n] = nums[i]

    for i in range(n):
        nums[i] = ans[i]  # update in-place
        

# Example usage
n, k = map(int, input().split())
v = list(map(int, input().split()))

rotate_left(v, k)
print("After left rotation:", v)

rotate_right(v, k)
print("After right rotation:", v)

File: Arrays/Easy/rotate_by_k_places.cpp
#include<bits/stdc++.h>
using namespace std;

void rotate_right(vector<int> &nums, int k)
{
      int n = nums.size();
      vector<int> ans(n);
      for (int i = 0; i < n; i++)
      {
            ans[(i + k) % n] = nums[i];
      }

      nums = ans;
}

void rotate_left(vector<int> &nums, int k)
{
      int n = nums.size();
      vector<int> ans(n);
      for (int i = 0; i < n; i++)
      {
            ans[(i - k + n) % n] = nums[i];
      }

      nums = ans;
}

int main()
{
      vector<int> v;
      int n,k;
      cin >> n>>k;

      v = vector<int>(n);
      for (int i = 0; i < n; i++)
      {
            cin >> v[i];
      }

      rotate_left(v,k);
      rotate_right(v,k);
}